interfaces/devdoc/string_utils.md

```diff
@@ -0,0 +1,61 @@
+`string_utils` requirements
+============
+
+## Overview
+
+`string_utils` is a collection of string utils. The set of requirements in this document is far from complete.
+
+## Exposed API
+
+```c
+/*produces a string as if printed by printf*/
+char* sprintf_char_function(const char* format, ...);
+
+/*produces a string as if printed by printf (will also verify arguments)*/
+#define sprintf_char(format, ...) (0?printf((format), ## __VA_ARGS__):0, sprintf_char_function((format), ##__VA_ARGS__))
+
+/*produces a string as if printed by wprintf*/
+wchar_t* sprintf_wchar_function(const wchar_t* format, ...);
+#define sprintf_wchar(format, ...) (0?wprintf((format), ##__VA_ARGS__):0, sprintf_wchar_function((format), ##__VA_ARGS__))
+
+MOCKABLE_INTERFACE(string_utils_printf,
+    /*produces a string as if printed by vprintf*/
+    FUNCTION(, char*, vsprintf_char, const char*, format, va_list, va),
+    /*produces a string as if printed by vwprintf*/
+    FUNCTION(, wchar_t*, vsprintf_wchar, const wchar_t*, format, va_list, va)
+)
+
+/*takes a FILETIME, returns a nice string representation of it*/
+MOCKABLE_FUNCTION(, char*, FILETIME_toAsciiArray, const FILETIME*, fileTime);
```

Comment:
>toAsciiArray [](http://example.com/codeflow?start=36&length=12)

an interesting name...should probably rename this and the new one so it is a little more clear what they do and what the difference is.

Reply:
addressed by [https://msazure.visualstudio.com/One/_workitems/edit/14684336](https://msazure.visualstudio.com/One/_workitems/edit/14684336)